### PR TITLE
dorion: init

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/pull/252627 per laziness

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AndersonTorres** [comment]
@knarkzel do the tests now.

**Comment by Nyabinary** [comment]
Should probably initiate the new version that should fix portable builds, 1.1.0.

**Comment by Nyabinary** [comment]
> @knarkzel do the tests now.

I can test by the way what exactly needs testing?

**Comment by LudovicoPiero** [comment]
Tested build. Can't join voice channel, stuck at RTC Connecting. Also v1.1.0 just dropped 2 days ago.

**Comment by AndersonTorres** [comment]
> > @knarkzel do the tests now.
> 
> I can test by the way what exactly needs testing?

`nix-shell -p nixpkgs-review --run "nixpkgs-review pr 253572"`

Inside folder `results/bin` you can find the binaries.

**Comment by Nyabinary** [comment]
> Tested build. Can't join voice channel, stuck at RTC Connecting. Also v1.1.0 just dropped 2 days ago.

Same problem here

**Comment by Nyabinary** [comment]
Maybe try the portable tar.gz in the new release instead of the .deb?

File: pkgs/applications/networking/instant-messengers/dorion/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    runHook postInstall
+  '';
+
+  meta = {
+    homepage = "https://github.com/SpikeHD/Dorion";
+    description = "Tiny alternative Discord client";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "dorion";
+    maintainers = with lib.maintainers; [ ];
+    platforms = lib.platforms.linux;
```
Comment (nagy) [inline pkgs/applications/networking/instant-messengers/dorion/default.nix:64]: I dont think this is correct. `lib.platforms.linux` includes things like aarch64-linux but we are only shipping amd64.

**Comment by Nyabinary** [comment]
Is the VC broken in the new release and/or with the portable pkg?

**Comment by AndersonTorres** [comment]
> Tested build. Can't join voice channel, stuck at RTC Connecting. 

Upstream issue.

**Comment by colemickens** [comment]
1.2.0 is out, claiming to not pollute my home dir :grin: 

```diff
-  version = "1.1.0";
+  version = "1.2.0";

   src = fetchurl {
     url = "https://github.com/SpikeHD/Dorion/releases/download/v${finalAttrs.version}/Dorion_${finalAttrs.version}_amd64.deb";
-    hash = "sha256-6PKH9OCfQrPc42CTDnwiB3tgd6KqfH/XjinrmfUaqBA=";
+    hash = "sha256-kpV/IUIzHgZYuFls+ThVnn/jGfZIwiCPJtB3mPHrf/U=";
   };
```

**Comment by AndersonTorres** [comment]
Please test this

**Comment by Nyabinary** [comment]
@AndersonTorres hash mismatch

**Comment by Nyabinary** [comment]
Also ```nix
ix/store/rq0ll828k6v62xicvmlmvaxm9qpg0ahf-dorion.drv' failed with exit code 1;
       last 10 log lines:
       > updateAutotoolsGnuConfigScriptsPhase
       > configuring
       > no configure script, doing nothing
       > building
       > no Makefile or custom buildPhase, doing nothing
       > glibPreInstallPhase
       > glibPreInstallPhase
       > glibPreInstallPhase
       > installing
       > mv: cannot stat 'usr/share/': No such file or directory
       For full logs, run 'nix log /nix/store/rq0ll828k6v62xicvmlmvaxm9qpg0ahf-dorion.drv'.
       ```

**Comment by Nyabinary** [comment]
Since it's an issue on Dorion end (it on their roadmap to fix it eventually), I think we can just update & merge this as it is?

**Comment by AndersonTorres** [comment]
I converted this to the zip file.

Now test!

**Comment by Nyabinary** [comment]
Compiles and builds for me! The client itself also works!

**Pull request merged**